From de812842f6aefc661bf261574e4962295a753cb8 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 17:55:27 -0800
Subject: [PATCH 1003/1009] MacPorts Only: Fix name of scan-view executable
 inside scan-build

http://trac.macports.org/ticket/35006

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit cf37a2cc4435f757228e928f5fbfb6cadd1b2e21)
---
 tools/scan-build/scan-build | 2 ++
 1 file changed, 2 insertions(+)

diff --git llvm_release_36/tools/clang/tools/scan-build/scan-build macports_release_36/tools/clang/tools/scan-build/scan-build
index d52d8f5..2a3d1db 100755
--- llvm_release_36/tools/clang/tools/scan-build/scan-build
+++ macports_release_36/tools/clang/tools/scan-build/scan-build
@@ -1728,6 +1728,8 @@ if (defined $OutputFormat) {
       Diag "Analysis run complete.\n";
       Diag "Viewing analysis results in '$HtmlDir' using scan-view.\n";
       my $ScanView = Cwd::realpath("$RealBin/scan-view");
+      if (! -x $ScanView) { $ScanView = Cwd::realpath("$RealBin/bin/scan-view"); }
+      if (! -x $ScanView) { $ScanView = Cwd::realpath("$RealBin/../scan-view/scan-view"); }
       if (! -x $ScanView) { $ScanView = "scan-view"; }
       exec $ScanView, "$HtmlDir";
     }
-- 
2.10.1

